src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -120,10 +107,12 @@ def close(self) -> None:
 
     def submit(self, message: Message[KafkaPayload]) -> None:
         assert not self.__closed
-        self.__messages_since_last_commit += 1
 
         payload = self._get_payload(message)
         self._produce_billing_outcomes(payload)
+        # TODO(markus): Perhaps call _bulk_commit directly here instead of
+        # manually accumulating offsets? Might be more in line with other
+        # consumers.
```

Comment:
I think you should call it directly and remove `_mark_commit_ready()` and `self.__ready_to_commit` dict as well.